### ISSUE TITLE
Enhancement for template including

### ISSUE THREAD

**Issue by donnior** [mbosecke/pebble#131]
Currently when we use `{% include "advertisement.html" %}` , the included template will have access to the same context (and therefore the same variables).  

For some scenes it would be better if we can pass variable context to the included template.  

For example when we rendering a collection of some models, we may need iteratively including one template and passing different context.

```
{% for article in articles %}
    {% include "article_detail.html", article %}
{% endfor %}
```

The `article_detail.html` maybe like: 
```
<h1>{{ article.title }}</h1>
{% for chapter in article.chapters %}
    <p>{{ chapter.content }}</p>
    ....
{% endfor %}
```

This included template is called '[Partial templates](http://guides.rubyonrails.org/layouts_and_rendering.html#using-partials)' in Ruby On Rails.  I think this feature will be very useful if Pebble can support it.

**Comment by ebussieres** [comment]
I think this was covered by issue #104

**Comment by mbosecke** [comment]
Yup, the new syntax which has already been committed and will be included in the next release (version 2.0) will look like this:

```
{% for article in articles %}
    {% include 'article_detail.html' with { 'article' : article } %}
{% endfor %}
```

Essentially, you can provide a map of extra variables to pass along. The template still has access to the original context PLUS the extra variables.
